SevenZip/SevenZipExtractor.cs

```diff
@@ -135,17 +136,18 @@ private void Init(Stream stream)
                     }
                 }
             }
-        }
-
+        }
+
         /// <summary>
         /// Initializes a new instance of SevenZipExtractor class.
         /// </summary>
         /// <param name="archiveStream">The stream to read the archive from.
+        /// <param name="leaveOpen">true to leave the wraped stream open after the ArchiveEmulationStreamProxy object is disposed; otherwise, false.</param>
         /// Use SevenZipExtractor(string) to extract from disk, though it is not necessary.</param>
         /// <remarks>The archive format is guessed by the signature.</remarks>
-        public SevenZipExtractor(Stream archiveStream)
+        public SevenZipExtractor(Stream archiveStream, bool leaveOpen = false)
```

Comment:
Pretty much just a suggestion, but wouldn't it be better to reverse this parameter - `bool disposeStream = true`?

Since the use of leaveOpen downstream is always `if  (!leaveOpen)` it would make for easier reading of the code, in my opinion at least.
The param comment would then be something like `Whether or not to dispose the stream after operation is complete.`.

Reply:
I have change it after @vasily-kirichenko review to make it more like [System.IO.StreamReader](https://docs.microsoft.com/en-US/dotnet/api/system.io.streamreader.-ctor?view=netcore-3.1#System_IO_StreamReader__ctor_System_IO_Stream_System_Text_Encoding_System_Boolean_System_Int32_System_Boolean_)